_posts/2020-05-18-mirror-kafka-eventhub.md

```diff
@@ -0,0 +1,225 @@
+---
+layout: post
+title:  "Mirroring from Apache Kafka to Azure Event Hub"
+date: 2020-05-18
+author: paolo_patierno
+---
+
+Let's imagine to have a software architecture based on Apache Kafka running on an on-premise Kubernetes installation and managed by the Strimzi operators.
+All the services are able to communicate with each other exchanging messages through a set of topics.
+At some point there is a new requirement to fullfil: all messages, sent to a specific topic, have to trigger some processing in the cloud via Microsoft Azure Functions already connected to an Azure Event Hub instance.
+It is a third party application already in place that we need to integrate with our own system.
+In order to avoid the producer applications to connect both the on-premise Apache Kafka cluster and to Azure Event Hub, which provides a Kafka protocol head, and sending each message twice, the best solution could be just mirroring the topic.
+This blog post is going to show how to use the Strimzi cluster operator for configuring and deploying Kafka Mirror Maker in order to mirror topics between two different systems like Apache Kafka and Azure Event Hub.
+
+<!--more-->
+
+### Integration overview
+
+Before starting to deploy the different components, let's take a look at how they should be integrated and the related overall architecture.
+
+![Overall architecture](/assets/images/posts/2020-05-18-kafka-mirror-eventhub.png)
+
+On one side there is a Kubernetes cluster where an Apache Kafka cluster is deployed and managed via the Strimzi operators as well as the Kafka Mirror Maker instance.
+An application, running on the same cluster, produces messages to an Apache Kafka topic and all the data are consumed by Kafka Mirror Maker in order to be mirrored to the remote Azure Event Hub instance.
+
+On the other side there is the Azure Event Hub namespace, on the Microsoft Azure Cloud, where an Event Hub is filled by the messages mirrored from the corresponding Apache Kafka topic.
+This messages trigger the Azure Functions platform to do the actual processing.
+For people who don't know about Azure Event Hub, it could be a little bit confusing and not really clear what an Event Hub "namespace" or just an Event Hub mean and how to compare them to an Apache Kafka cluster and topic; it will be clarified in the Event Hub related paragraph of this blog post.
```

Comment:
You're saying "You might now know what X means, but tough, I'm not going to explain it here. You'll need to be interested enough to read about that later". That's really unhelpful to those readers who don't know about X already. What are your reasons for doing it? I think it would be better to have a paragraph or section where you explain it, and encourage those readers who already know to skip to the next part.

Reply:
Maybe I can move the section where I explain it from the EH paragraph to here?

Reply:
TBH I think you could just remove this sentence.

Reply:
Yeah, that's a good idea. They will learn more in the EH paragraph if they don't know what it is and how it works.